packages/core/frontend/sirius-components-core/src/selection/SelectionContext.tsx

```diff
@@ -41,7 +41,10 @@ export const SelectionContextProvider = ({ initialSelection, children }: Selecti
         .map((entry) => entry.id)
         .sort()
         .join(':');
-      if (prevSelectionKey !== newSelectionKey) {
+      if (
+        prevSelectionKey !== newSelectionKey ||
+        selection.entries.find((entry) => entry.kind.startsWith('siriusComponents://representation?type='))
+      ) {
```

Comment:
I'm afraid we will have to remove the previous "improvement" since we will need to consider more than just representations :/